front-end-web/src/Components/Pages/ProfilePage.tsx

```diff
@@ -0,0 +1,184 @@
+import { NotificationMessage } from "../../@Types";
+declare interface ProfileProps {
+  openPage: boolean;
+  handleToggleProfile: () => void;
+
+  handleNewNotification: (NewNotification: NotificationMessage) => void;
+}
+
+/**
+ * TODO: Add Documentation
+ */
+
+let newEmail: string = "";
+let newPassword: string = "";
+let newPasswordCheck: string = "";
+let newName: string = "";
+let currentName: string = "";
+let currentEmail: string = "";
+
+firebaseApp.auth();
+
+const ProfilePage: React.FunctionComponent<ProfileProps> = ({
+  openPage,
+  handleToggleProfile,
+  handleNewNotification
+}) => {
+  //const [open, setOpen] = React.useState(false);
+
+  // const handleOpen = () => {
+  //   setOpen(true);
+  //};
+
+  const handleChangeEmail = () => {
+    const currentUser = firebase.auth().currentUser;
+    if (currentUser) {
+      currentUser.updateEmail(newEmail);
+      handleNewNotification({
+        message: "Email changed successfully.",
+        type: "success",
+        open: true
+      });
+    }
+  };
+
+  const handleChangePass = () => {
+    const currentUser = firebase.auth().currentUser;
+    if (currentUser) {
+      currentUser.updatePassword(newPassword);
+      handleNewNotification({
+        message: "Password changed successfully",
+        type: "success",
+        open: true
+      });
+    }
+  };
+  const [openPass, setOpenPass] = React.useState(false);
+  const handleOpenPasswordDia = () => {
+    handleToggleProfile();
+    setOpenPass(true);
+  };
+  const handleClosePasswordDia = () => {
+    setOpenPass(false);
+  };
+
+  const handleNewPassword = () => {
+    if (newPassword === newPasswordCheck) {
+      handleChangePass();
+      handleClosePasswordDia();
+    } else {
+      handleNewNotification({
+        message: "Passwords do not match.",
+        type: "error",
+        open: true
+      });
+    }
+  };
+
+  const handleChangeName = () => {
+    const currentUser = firebase.auth().currentUser;
+    if (currentUser) {
+      currentUser.updateProfile({ displayName: newName });
+      handleNewNotification({
+        message: "Display name changed successfully.",
+        type: "success",
+        open: true
+      });
+    }
+  };
+  const currentUser = firebase.auth().currentUser;
+  if (currentUser) {
+    if (currentUser.displayName !== null) {
+      currentName = currentUser.displayName;
+    }
+    if (currentUser.email !== null) {
+      currentEmail = currentUser.email;
+    }
+  }
+  return (
+    <Fragment>
+      <Dialog open={openPage} aria-labelledby="form-dialog-title">
+        <DialogTitle id="form-dialog-title">User Profile</DialogTitle>
+        <DialogContent>
+          <DialogContentText></DialogContentText>
+          <TextField
+            margin="dense"
+            helperText="Display Name"
+            id="name"
+            fullWidth
+            defaultValue={currentName}
+            onChange={e => (newName = e.target.value)}
+          />
+        </DialogContent>
```

Comment:
`DialogContent` should include all of the content, including the buttons and email reset. I would move the closing tag until right before `DialogActions` is opened.